### DISCUSSION TITLE
Difference with isort black profile - ruff isort groups "from x import a, z" even if there's a "from x import b as c"

### DISCUSSION THREAD

**Discussion by notatallshaw-gts** [astral-sh/ruff#9617] [General]
I am looking at moving from isort to ruff's isort implementation. I appreciate it's an approximation but I am trying to understand if this change is intentional or not.

isort with profile=black will sort like this:

```python
from foo import a
from foo import c as b
from foo import d
```

Where as the ruff isort approximation will sort like this:

```python
from foo import a, d
from foo import c as b
```

Is this intentional? And is there a related setting? (I did look and try different things, but this didn't change). Has there been other discussions on this? (Again I did look).

FYI, I don't feel like I prefer one behavior over the other, but I would like to make a relatively minimal diff switching from isort to ruff.

**Comment by charliermarsh** [comment]
I believe this was intentional, we avoid breaking up the non-`as` imports into multiple import statements. You can set `combine-as-imports` to avoid splitting them at all, but isort doesn't default to that behavior, so we don't either.

**Reply by notatallshaw-gts** [reply]
Appreciate confirming it is intended!

Unfortunately `combine-as-imports=true` seems to be more disruptive compared to isort with profile=black, than not setting it. For this isorted import list:

```python
from foo import a
from foo import c as b
from foo import d
```

It gets transformed to this:

```python
from foo import a, c as b, d
```

Which for my codebase jumps the number of affected Python files from ~5% to ~18%. So I'll certainly stick with the 5%.
